### ISSUE TITLE
Prepare gcc-14 for tizen gbs build

### ISSUE THREAD

**Issue by chunseoklee** [Samsung/ONE#13133]
# Background

- tizen gbs will update gcc version to v.14.1.0 (https://review.tizen.org/gerrit/#/q/branch:tizen_base+topic:gcc-14)
- onert build break with gcc v14.1

**Comment by chunseoklee** [comment]
local build guide : 

gbs.conf
```
[general]
profile = profile.tizen_unified_standard
buildroot = ~/GBS-ROOT-TOOLCHAIN
 
[profile.tizen_unified_standard]
repos = repo.base-standard, repo.unified-standard, repo.base-standard-debug, repo.unified-standard-debug
 
[repo.base-standard]
url=https://download.tizen.org/snapshots/TIZEN/Tizen/Tizen-Base-Toolchain/reference/repos/standard/packages/
 
[repo.base-standard-debug]
url=https://download.tizen.org/snapshots/TIZEN/Tizen/Tizen-Base-Toolchain/reference/repos/standard/debug/
 
[repo.unified-standard]
url=https://download.tizen.org/releases/daily/TIZEN/Tizen/Tizen-Unified-Toolchain/tizen-unified-toolchain_20240531.025943/repos/standard/packages/
 
[repo.unified-standard-debug]
url=https://download.tizen.org/releases/daily/TIZEN/Tizen/Tizen-Unified-Toolchain/tizen-unified-toolchain_20240531.025943/repos/standard/debug/
```

build command : 

```
gbs -c ~/.gbs_toolchain.conf build -A {i586|x86_64|armv7l|armv7hl|aarch64|riscv64} --include-all
```

**Comment by chunseoklee** [comment]
With above gbs.conf

```
$ gbs -v -c ~/gbs_gcc14.conf build -A armv7l --include-all

[   66s] -- Configure LUCI-COMPUTE - Done
[   66s] -- Configure MIO-CIRCLE06
[   66s] -- Build FLATBUFFERS-2.0 from /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/infra/nncc/../../externals/FLATBUFFERS-2.0
[   70s] CMake Error at /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/infra/cmake/modules/ExternalBuildTools.cmake:81 (message):
[   70s]   FLATBUFFERS-2.0 Package: Installation failed (check
[   70s]   '/home/abuild/rpmbuild/BUILD/nnfw-1.25.0/build/nncc/overlay/FLATBUFFERS-2.0.log'
[   70s]   for details)
[   70s] Call Stack (most recent call first):
[   70s]   /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/infra/cmake/packages/FlatBuffers-2.0/FlatBuffersConfig.cmake:30 (ExternalBuild_CMake)                                                                                                                    [   70s]   /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/infra/cmake/packages/FlatBuffers-2.0/FlatBuffersConfig.cmake:41 (_FlatBuffers_build)                                                                                                                     [   70s]   CMakeLists.txt:38 (find_package)
[   70s]   /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/compiler/mio-circle06/CMakeLists.txt:1 (nnas_find_package)
[   70s]
[   70s]
[   70s] -- Configuring incomplete, errors occurred!
[   70s] See also "/home/abuild/rpmbuild/BUILD/nnfw-1.25.0/build/nncc/CMakeFiles/CMakeOutput.log".
[   70s] error: Bad exit status from /var/tmp/rpm-tmp.UjjdBe (%build)
[   70s]
[   70s]
```

```
[ 83%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/java_generator.cc.o
[ 86%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/python_generator.cc.o
In file included from /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/include/flatbuffers/flatbuffers.h:21,
                 from /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/src/idl_gen_text.cpp:19:
/home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/include/flatbuffers/stl_emulation.h: In member function 'flatbuffers::span<T, Extent>& flatbuffers::span<T, Extent>::operator=(const flatbuffers::span<T, Extent>&)':
/home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/include/flatbuffers/stl_emulation.h:549:12: error: assignment of read-only member 'flatbuffers::span<T, Extent>::count_'
  549 |     count_ = other.count_;
      |     ~~~~~~~^~~~~~~~~~~~~~
In file included from /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/include/flatbuffers/flatbuffers.h:21,
                 from /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/src/idl_gen_text.cpp:19:
/home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/include/flatbuffers/stl_emulation.h: In member function 'flatbuffers::span<T, Extent>& flatbuffers::span<T, Extent>::operator=(const flatbuffers::span<T, Extent>&)':
/home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/FLATBUFFERS-2.0/include/flatbuffers/stl_emulation.h:549:12: error: assignment of read-only member 'flatbuffers::span<T, Extent>::count_'
  549 |     count_ = other.count_;
      |     ~~~~~~~^~~~~~~~~~~~~~
```

```
  FLATBUFFERS_CONSTEXPR_CPP14 span &operator=(const span &other)
      FLATBUFFERS_NOEXCEPT {
    data_ = other.data_;
    count_ = other.count_;
  }
```

**Comment by chunseoklee** [comment]
patch ? : https://github.com/google/flatbuffers/commit/a6f41944899e65b38ec45ee82f6840248f06b471

**Comment by chunseoklee** [comment]
https://github.com/google/flatbuffers/pull/7227/

**Comment by chunseoklee** [comment]
public tizen patch : https://review.tizen.org/gerrit/gitweb?p=platform/core/ml/nnfw.git;a=commit;h=9d8027e1a93c4cf3e717cc0c3db1f55dd1db58d5

**Comment by chunseoklee** [comment]
Heard that it failed on riscv64 build:

```
@이춘석 님.
[  125s] /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/CPUINFO/src/init.c:19:24: warning: 'init_guard' defined but not used [-Wunused-variable]
[  125s]    19 |         static pthread_once_t init_guard = PTHREAD_ONCE_INIT;
[  125s]       |                               ^~~~~~~~~~
[  125s] In file included from /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/CPUINFO/src/cpuinfo/internal-api.h:11,
[  125s]                  from /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/CPUINFO/src/api.c:5:
[  125s] /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/CPUINFO/src/api.c: In function 'cpuinfo_get_current_processor':
[  125s] /home/abuild/rpmbuild/BUILD/nnfw-1.25.0/externals/CPUINFO/src/api.c:319:23: error: implicit declaration of function 'syscall' [-Wimplicit-function-declaration]
[  125s]   319 |                 if CPUINFO_UNLIKELY(syscall(__NR_getcpu, &cpu, NULL, NULL) != 0) {
```
